src/Lib.P2P/PubSub/TopicManager.cs

```diff
@@ -0,0 +1,140 @@
+#region LICENSE
+
+/**
+* Copyright (c) 2019 Catalyst Network
+*
+* This file is part of Catalyst.Node <https://github.com/catalyst-network/Catalyst.Node>
+*
+* Catalyst.Node is free software: you can redistribute it and/or modify
+* it under the terms of the GNU General Public License as published by
+* the Free Software Foundation, either version 2 of the License, or
+* (at your option) any later version.
+*
+* Catalyst.Node is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+* GNU General Public License for more details.
+*
+* You should have received a copy of the GNU General Public License
+* along with Catalyst.Node. If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#endregion
+
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Lib.P2P.PubSub
+{
+    /// <summary>
+    ///   Maintains the sequence of peer's that are interested in a topic.
+    /// </summary>
+    public class TopicManager
+    {
+        private static readonly IEnumerable<Peer> Nopeers = Enumerable.Empty<Peer>();
+
+        private ConcurrentDictionary<string, HashSet<Peer>> _topics = new ConcurrentDictionary<string, HashSet<Peer>>();
+
+        /// <summary>
+        ///   Get the peers interested in a topic.
+        /// </summary>
+        /// <param name="topic">
+        ///   The topic of interest or <b>null</b> for all topics.
+        /// </param>
+        /// <returns>
+        ///   A sequence of <see cref="Peer"/> that are interested
+        ///   in the <paramref name="topic"/>.
+        /// </returns>
+        public IEnumerable<Peer> GetPeers(string topic)
+        {
+            if (topic == null) return _topics.Values.SelectMany(v => v);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/manual/catalyst-network/Catalyst.Framework/pullRequest?prid=4832783)